example-deck.gl.md

```diff
@@ -0,0 +1,400 @@
+  ],
+  "type": "FeatureCollection"
+```
+
+What we get here is a standardized [GeoJSON FeatureCollection](https://geojson.org/).
+GeoJSON objects are often used for handling geographic data structures and are supported by a variety of different tools.
+
+## Requesting Data
+
+After we now know how the data is structured and where we can find it, we have to think about how we get it.
+Luckily enough, the deck.gl object we are using later is able to use the URL and handle the request itself.
+So we only have to construct the URL:
+
+```javascript
+// A polygon describing the vague form of Germany
+const GERMANY = 'polygon=9.921906365609232,54.983104153048025,' +
+    '9.9395797054529,54.596641954153256,' +
+    '10.950112338920519,54.363607082733154,' +
+    …
+    '8.526229282270208,54.96274363872516,' +
+    '9.282048780971136,54.83086538351631,' +
+    '9.921906365609232,54.983104153048025';
+
+// API URL
+const API_URL = 'https://demo.emissions-api.org/api/v1/geo.json?' +
+    GERMANY + '&begin=2019-02-04&end=2019-02-06';
+```
+
+## Plotting Data
+
+We are using the [Mapbox](https://www.mapbox.com/) support, deck.gl offers, and use Mapbox as the base map for our plot.
+Therefore we have to include the Mapbox script and stylesheet:
+
+```html
+<script src='https://api.tiles.mapbox.com/mapbox-gl-js/v0.53.0/mapbox-gl.js'></script>
+<link href='https://api.tiles.mapbox.com/mapbox-gl-js/v0.53.0/mapbox-gl.css' rel='stylesheet' />
+```
+
+and generate a [Mapbox Token](https://docs.mapbox.com/help/how-mapbox-works/access-tokens/) for later use.
+
+Deck.gl itself offers a JavaScript scripting interface,
+see [DeckGL (Scripting Interface)](https://deck.gl/#/documentation/deckgl-api-reference/scripting-interface/deckgl).
+To use this we additional include:
+
+```html
+<script src="https://unpkg.com/deck.gl@latest/dist.min.js"></script>
+```
+
+We also need to define a HTML-container used from deck.gl to work on:
+
+```html
+<style type="text/css">
+      #container {width: 100%; height: 500px;}
+</style>
+<div id="container"></div>
+
+<script src="https://unpkg.com/deck.gl@latest/dist.min.js"></script>
+```
+
+Now we can start to write our JavaScript code.
+
+Deck.gl will combine points in the same hexagon to an array and let the user define how the elevation and color of this hexagon is calculated.
+We choose to use the average of all points in a hexagon for this.
+So we need a small helper function:
+
+```javascript
+// Helper function to calculate the average value
+function average(numbers) {
+    let total = 0;
+    for (let i = 0; i < numbers.length; i += 1) {
+        total += numbers[i];
+    }
+    return total / numbers.length;
+}
+```
+
+Now let's define the visualization layer we have seen above:
+
+```javascript
+new deck.HexagonLayer({
+    extruded: true,
+    radius: 5000,
+    data: API_URL,
+    dataTransform: d => d.features,
+    elevationScale: 100,
+    getColorValue: points => average(points.map(d => d.properties.carbonmonoxide)),
```

Comment:
If we do a mapping anyway, maybe move this into the average method?

Reply:
Because I did not know how :smile:

Reply:
JavaScript arrow functions are basically just Python lambdas.
So we should be able to just set `getColorValue` to a function accepting points.